pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/ArrayLiteralTransformFunction.java

```diff
@@ -218,7 +218,33 @@ public int[][] transformToIntValuesMV(ValueBlock valueBlock) {
     int[][] intArrayResult = _intArrayResult;
     if (intArrayResult == null || intArrayResult.length < numDocs) {
       intArrayResult = new int[numDocs][];
-      Arrays.fill(intArrayResult, _intArrayLiteral);
+      if (_intArrayLiteral != null) {
```

Comment:
We want to set `_intArrayLiteral` only once, same for other places
```suggestion
      if (_intArrayLiteral == null) {
        switch (_dataType) {
          case LONG:
            ...
        }
      }
      Arrays.fill(intArrayResult, _intArrayLiteral);
```